File: twosum.py
# Given a list of unique numbers and a target, return indices of the two numbers such 
# that they add up to a specific target.

def two_sum(nums, target):

    index_and_num = {}

    for index, num in enumerate(nums):
        index_and_num[num] = index
        
        complement = target - num
        if complement in index_and_num:
            return [index_and_num[num], index_and_num[complement]]

    raise Exception("No two numbers add up to the target")


print two_sum([1, 2, 3, 4], 7)

File: twosum.js
var twoSum = function(nums, target) {
    
    var index_and_nums = {};
    var final = []

    for (i=0; i <= nums.length; i++){

        console.log(nums[i])

        index_and_nums[nums[i]] = i;

        console.log(index_and_nums)

        var complement = target - nums[i];
        console.log(complement)

        if (complement in index_and_nums == true) {
            final.push(index_and_nums[nums[i]])
            final.push(index_and_nums[complement])
            return final
        }
    }
}